### PR TITLE
Anne/cpp

### PR DESCRIPTION
**Pull request by EnnaDelfen**
Hi guys

Here's a branch with a lot of cpp/F90 cleaning in EOS/Transport and Reaction dir.
I have made substantial modifications to the Null Reaction folder, and the Makefile system is probably not bug-free. So please, go ahead and try this branch with your applications and give me a report so I can make it even better ! Thanks

### CODE REVIEW

**Comment by EnnaDelfen** [comment]
Note that I'm in the process of fixing the regtests, they are not expected to work right now

**Comment by drummerdoc** [comment]
Ahh, soooo much more consistent! 

I'm only a little worried about case insensitive OSs like OSX getting confused between ``GET_MW`` and ``get_mw`` - can we rename one of them to ``molecular_weight`` or something (same for Imw)?

**Comment by ldowen** [comment]
Looks good ! I have just a few things that might need to be looked at:

1. It looks like you missed renaming some things in Simple/Transport.H and Simple/TransportParam.cpp. Specifically, things like changing `fitlam` to `eos_fitlam` etc.
2. It seems like a different prefix for the transport variables other than `eos_` would be better. Maybe something like `trans_`?
3. `array_size` and `fit_length` should just be defined locally at the top of TransportParams.cpp in `init()`

**Comment by drummerdoc** [comment]
As for #2, the intention on a later pass is to replace EOS with PPhys, or something similar, which should reflect the appropriate level of generality for all things under PelePhysics.  Just trying to limit an already huge merge.

**Comment by drummerdoc** [comment]
I would also like to *finally* get rid of "actual_" from the filenames as well.  that is a leftover from a redundant level of layering which this reorg nicely clipped out.  But we'll do that renaming on a different merge as well.  (Also, remove "CK" from the names of functions generated by FUEGO, plus probably host of other longstanding annoyances... :) )

**Comment by EnnaDelfen** [comment]
Argh no Landon is right, I was modifying Transport var and thought I was in EOS... I'll blame the late hour. I'll modify that, it's too icky.
Yes the actual are going to go. Definitely the next pull request, but as you said Marc, trying to do things incrementally.
Working on the regtests right now.

**Comment by EnnaDelfen** [comment]
@drummerdoc  the regtests are good to go. The only happy green guy is the TranEval-2d-gamma test because the Constant Transport model is very dumb and nothing I did changed anything there. I backtraced my changes one by one for TranEval-2d-LD and both EOS test cases and the diff between the dev branch and this branch are solely due to the changes that I introduced in the Ru and Patm constants throughout PP. The ReactEval test case I just looked it in the eye and trust it's OK -and if not, i'll find out soon enough since I spend my life playing with it.

**Comment by drummerdoc** [comment]
I want to understand... What does it mean that "the Constant Transport model is very dumb"

**Comment by EnnaDelfen** [comment]
it means the values for transport coeffs are simply provided by the user. So nothing I changed( the values of the universal constants ) affected this model

**Review by jrood-nrel — CHANGES_REQUESTED** [review]
I had to make these changes to get our C++ PeleC to compile:
```
diff --git a/Support/Fuego/Mechanism/Models/LiDryer/CMakeLists.txt b/Support/Fuego/Mechanism/Models/LiDryer/CMakeLists.txt
index a4091fe..e8e315a 100644
--- a/Support/Fuego/Mechanism/Models/LiDryer/CMakeLists.txt
+++ b/Support/Fuego/Mechanism/Models/LiDryer/CMakeLists.txt
@@ -2,7 +2,7 @@ target_sources(${pele_exe_name}
    PRIVATE
       chemistry_file.H
       mechanism.h
-      LiDryer.cpp
+      mechanism.cpp
    )
 
 target_include_directories(${pele_exe_name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
diff --git a/Support/Fuego/Mechanism/Models/Null/CMakeLists.txt b/Support/Fuego/Mechanism/Models/Null/CMakeLists.txt
index a52da49..98c005b 100644
--- a/Support/Fuego/Mechanism/Models/Null/CMakeLists.txt
+++ b/Support/Fuego/Mechanism/Models/Null/CMakeLists.txt
@@ -1,7 +1,7 @@
 target_sources(${pele_exe_name}
    PRIVATE
       mechanism.h
-      null.cpp
+      mechanism.cpp
    )
 
 target_include_directories(${pele_exe_name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
diff --git a/Support/Fuego/Mechanism/Models/air/CMakeLists.txt b/Support/Fuego/Mechanism/Models/air/CMakeLists.txt
index e7e6613..e81b74e 100644
--- a/Support/Fuego/Mechanism/Models/air/CMakeLists.txt
+++ b/Support/Fuego/Mechanism/Models/air/CMakeLists.txt
@@ -2,7 +2,7 @@ target_sources(${pele_exe_name}
    PRIVATE
       chemistry_file.h
       mechanism.h
-      air.cpp
+      mechanism.cpp
    )
 
 target_include_directories(${pele_exe_name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
diff --git a/Support/Fuego/Mechanism/Models/decane_3sp/CMakeLists.txt b/Support/Fuego/Mechanism/Models/decane_3sp/CMakeLists.txt
index 338dfe9..e81b74e 100644
--- a/Support/Fuego/Mechanism/Models/decane_3sp/CMakeLists.txt
+++ b/Support/Fuego/Mechanism/Models/decane_3sp/CMakeLists.txt
@@ -2,7 +2,7 @@ target_sources(${pele_exe_name}
    PRIVATE
       chemistry_file.h
       mechanism.h
-      decane_3sp.cpp
+      mechanism.cpp
    )
 
 target_include_directories(${pele_exe_name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
diff --git a/Support/Fuego/Mechanism/Models/drm19/CMakeLists.txt b/Support/Fuego/Mechanism/Models/drm19/CMakeLists.txt
index 69c9ca9..e8e315a 100644
--- a/Support/Fuego/Mechanism/Models/drm19/CMakeLists.txt
+++ b/Support/Fuego/Mechanism/Models/drm19/CMakeLists.txt
@@ -2,7 +2,7 @@ target_sources(${pele_exe_name}
    PRIVATE
       chemistry_file.H
       mechanism.h
-      drm19.cpp
+      mechanism.cpp
    )
 
 target_include_directories(${pele_exe_name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
diff --git a/Support/Fuego/Mechanism/Models/heptane_3sp/CMakeLists.txt b/Support/Fuego/Mechanism/Models/heptane_3sp/CMakeLists.txt
index 769c951..e81b74e 100644
--- a/Support/Fuego/Mechanism/Models/heptane_3sp/CMakeLists.txt
+++ b/Support/Fuego/Mechanism/Models/heptane_3sp/CMakeLists.txt
@@ -2,7 +2,7 @@ target_sources(${pele_exe_name}
    PRIVATE
       chemistry_file.h
       mechanism.h
-      heptane_3sp.cpp
+      mechanism.cpp
    )
 
 target_include_directories(${pele_exe_name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
diff --git a/Transport/Simple/Transport.H b/Transport/Simple/Transport.H
index 13ac570..07512fc 100644
--- a/Transport/Simple/Transport.H
+++ b/Transport/Simple/Transport.H
@@ -52,7 +52,7 @@ comp_pure_bulk(amrex::Real Tloc, amrex::Real* muloc, amrex::Real* xiloc)
       cvkint[i] = cvk[i] * trans_wt[i] / Ru - 1.50;
       cvkrot[i] = 1.0;
     } else {
-      cvkint[i] = cvk[i] * wt[i] / Ru - 1.50;
+      cvkint[i] = cvk[i] * trans_wt[i] / Ru - 1.50;
       cvkrot[i] = 1.5;
     }
   }
diff --git a/Transport/Simple/TransportParams.cpp b/Transport/Simple/TransportParams.cpp
index cc9ad47..4f70a2b 100644
--- a/Transport/Simple/TransportParams.cpp
+++ b/Transport/Simple/TransportParams.cpp
@@ -35,7 +35,7 @@ init()
     amrex::The_Managed_Arena()->alloc(sizeof(amrex::Real) * array_size));
   trans_sig = static_cast<amrex::Real*>(
     amrex::The_Managed_Arena()->alloc(sizeof(amrex::Real) * array_size));
-  dip = static_cast<amrex::Real*>(
+  trans_dip = static_cast<amrex::Real*>(
     amrex::The_Managed_Arena()->alloc(sizeof(amrex::Real) * array_size));
   trans_pol = static_cast<amrex::Real*>(
     amrex::The_Managed_Arena()->alloc(sizeof(amrex::Real) * array_size));
```
I can't get our Fortran version of PeleC to compile at all with these changes. I'm not sure I can sort out what I need to change. It looks like all the fortran here moved and some of the f90 files were deleted. Can you summarize what I need to do to get this repo to work the way it once did for the PeleC `development` branch?

**Comment by jrood-nrel** [comment]
I have spent at least 3 hours trying to adapt the old PeleC code to these changes and I am unable to. I am worried there is far too much going on in this pull request.

**Comment by jrood-nrel** [comment]
What if we keep a snapshot of the current PelePhysics for the fortran in PeleC and then remove the fortran entirely with this pull request? Or split the fortran and C++ at a higher level like we've done in our PeleC C++ branch to allow for less modification of PeleC?

**Comment by EnnaDelfen** [comment]
@jrood-nrel  I just pushed one modif in this PelePhysics branch to make it work with the Anne/cpp branch of PeleC. So if you use the current Anne/cpp up-to-date branch in PP and the corresponding one in PeleC, the PMF regtest will build and run fine. To see specifically what I changed in the PeleC dev branch (upon which Anne/cpp is based) I'll refer you to the pull request I opened in PeleC !
Note that a compare will give you differences. This is because I updated all R and Patm constants throughout PelePhysics for them to be the same everywhere.

**Comment by jrood-nrel** [comment]
@EnnaDelfen It looks like the `mod_fuego.F90`, and essentially Fuego is now required. What about a case like `TG` where there is no reactions and no Fuego? It seems like the make files need more logic to account for this.

**Comment by EnnaDelfen** [comment]
So now there will always be a reaction dir, and it will always be fuego. It makes more sense to put all chemistry in the same place, that network was an unecessary FORTRAN layer, and in the idea of moving to cpp soon it made sense to condense all chem stuff in one module. So when before you specified reaction_dir= null and chem_dir= nothing you would put reaction_dir= fuego and chen_dir= null in your gnu makefile. There might be some adjustments, but they should honestly be minor. Either you try and let me know or point me to such a test case so I can make the necessary adjustments !

**Comment by EnnaDelfen** [comment]
Ok the TG test case is running now.

**Comment by ldowen** [comment]
I just pushed back some minor bug fixes for the Simple transport.

**Comment by EnnaDelfen** [comment]
Thanks Landon.
@jrood-nrel do these changes seem acceptable to you ? If not, can you formulate a plan to move forward ? Thanks !

**Review by jrood-nrel — APPROVED** [review]
Ok I decided to delete my CMake from this repo and I will just manage it in PeleC like I've done in the past. Thanks for helping update PeleC for these changes @EnnaDelfen . I am fine merging this. I just have one more question. Why is there `YT2E` that takes mass fractions but doesn't use them?

**Comment by ldowen** [comment]
I think that is also the case for `YT2H`. It looks like those should probably be called `T2Ei` and `T2Hi` instead. We should probably change those, unless there is something I am missing.

**Comment by jrood-nrel** [comment]
I only noticed because I had to change `T2E` to `YT2E` in my PeleC `cpp` branch.

**Comment by drummerdoc** [comment]
Landon's right, according to the naming scheme.  The one that takes mass fractions should return the bulk quantities.  Otherwise, they should have an "I" on the end of their name.

**Comment by ldowen** [comment]
I can correct those right now and push it back upstream. Is that alright?

**Comment by EnnaDelfen** [comment]
Think I called this YT2H because a) i didn't realize they where for each spec and b) I added Y because you need it for GammaLaw and if you want a piece of code that's general enough to get H whatever the eos then you need to include the max number of args for each model. But we can certainly change that for Fuego eos

**Comment by ldowen** [comment]
Also, did we want to change the names for `GET_MW` and `GET_IMW`? To avoid issues with capitalizations? Also, I think we can get away with changing them for both Fuego and GammaLaw if we just hard-code in the molecular weight. Can I make some changes and push them back? If we hate them, we can get rid of them but I think they might clean up  few things.

**Comment by jrood-nrel** [comment]
Oh I see the `YT2H` in GammaLaw. That makes sense to me then.

**Comment by EnnaDelfen** [comment]
Sorry Landon I was too quick sure make the modifs !

**Comment by ldowen** [comment]
The GammaLaw uses the same molecular weights for all species, so Wbar should always be a constant value. So I don't think the mass fractions are needed in those functions. I will make some changes now!

**Comment by drummerdoc** [comment]
As a head's up, some of the function names will definitely change when you add the non-ideal stuff, since the bulk functions typically also require rho as input (the more general form from thermodynamic arguments).  In those cases, we'll have to add an "R" in the name (and put density in the correct argument slot).  But, when we do that, the compiler will tell us where we have to make the required changes, and we can do that later.

**Comment by ldowen** [comment]
Ok. I pushed my changes upstream.

**Comment by EnnaDelfen** [comment]
Ok let me just check in Fuego that the values of PATM and RUC that you put in EOS are consistent. I'll do that later today and then, unless I hear anything further, I'll merge.
Thanks y'all

**Comment by ldowen** [comment]
I copied the values in Fuego/EOS.H to GammaLaw/EOS.H. So they should be correct.

**Comment by EnnaDelfen** [comment]
I meant in the chemistry Fuego, the python script that generates the cpp for the chemistry ! There's use of those RU/RUC/PATM everywhere

**Comment by ldowen** [comment]
Oh I see!

**Comment by ldowen** [comment]
I foolishly left a merge conflict in my push. Sorry!

**Comment by jrood-nrel** [comment]
This looks good to me. Seems to be in mostly working order for PeleC from what I have tried. Can we merge this?

**Comment by EnnaDelfen** [comment]
Almost :) I need to check a couple things, but haven't taken the time yet. I'll merge before tonight !
